compiler/src/iree/compiler/Codegen/Dialect/PCF/IR/PCFTypes.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Dialect/PCF/IR/PCFTypes.h"
+#include "iree/compiler/Codegen/Dialect/PCF/IR/PCFAttrs.h"
+#include "llvm/ADT/TypeSwitch.h"
+#include "llvm/Support/LogicalResult.h"
+
+#define GET_TYPEDEF_CLASSES
+#include "iree/compiler/Codegen/Dialect/PCF/IR/PCFTypes.cpp.inc" // IWYU pragma: keep
+
+namespace mlir::iree_compiler::IREE::PCF {
+
+//===----------------------------------------------------------------------===//
+// #pcf.sref<...>
+//===----------------------------------------------------------------------===//
+
+Type ShapedRefType::parse(AsmParser &parser) {
+  if (parser.parseLess()) {
+    return {};
+  }
+
+  llvm::SmallVector<int64_t> shape;
+  mlir::Type elementType;
+  Attribute scope;
+
+  auto shapeLoc = parser.getCurrentLocation();
+  if (mlir::failed(parser.parseDimensionList(shape))) {
```

Comment:
```suggestion
  if (failed(parser.parseDimensionList(shape))) {
```
also elsewhere below